### PR TITLE
create hypothesis and cost classes

### PR DESCRIPTION
**Pull request by ckaminer**
Hypothesis Function:
- hypothesis class acts as simple linear formula
- takes intercept and slope as arguments
- unit-tested

Cost Function:
- cost function calculates cost given training set
- cost returns normalized sum of squares
- uses hypothesis instance in cost function
- practice training set at the bottom is real data taken from the runner

Runner:
- Update runner to produce the training set I plan on using
- Runner now returns array of hashes with i/o keys
- input represents total team hits at home in a season
- output represents total team runs at home in a season

Next Steps:
- Now that the data, hypothesis, and costs are set up I will actually train the data
- Will make a new "Regression" class that the runner will call on to train the retrieved data

### CODE REVIEW

**Review by icorson3 — COMMENTED** [review]
I think that your PR is very easy to understand even though I have never see python code. It looks a lot like ruby. Changing "run_counts" to "training_set" made that more explicit.

**Comment by AnnaCW** [comment]
No suggestions, but it's interesting to see some Python. Is the 'try/except' statement like an 'if/else' checking for both runs and hits and incrementing accordingly? What would trigger the 'n/a'?

File: cost_function.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import math
+from hypothesis_function import HypothesisFunction
+
+class CostFunction:
+    def __init__(self, training_set):
```
Comment (lucyfox4131) [inline cost_function.py:5]: Looks easier then spelling initialize

**Comment by ckaminer** [comment]
@AnnaCW the try/except is different from an if/else (Python does have an if/else too).  The try/except is to deal with errors when enumerating.  In this case it tries to execute the 'try block' but some of the games in the array don't have a game id that can be found in the team stats library.  So if when it tries to find said game and there is an error, it won't break the loop.  It might be more explicit if I change the "n/a" to `runs + 0` or just a more explicit message.

File: cost_function.py (hunk 1 above)
Comment (ghost) [reply cost_function.py:5]: @lucyfox4131 That's really funny! I think the underscore syntax is a little strange, but I guess it takes some getting used to. Overall, Python looks pretty clean

**Pull request merged**